src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,120 @@
+package mate.academy.dao;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+import mate.academy.util.ConnectionUtil;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final String CREATE = "INSERT INTO books (title, price) VALUES (?, ?)";
+    private static final String FIND_BY_ID = "SELECT * FROM books WHERE id = ?";
+    private static final String FIND_ALL = "SELECT * FROM books";
+    private static final String UPDATE = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+    private static final String DELETE_BY_ID = "DELETE FROM books WHERE id = ?";
+
+    @Override
+    public Book create(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement statement = connection.prepareStatement(CREATE,
+                     Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new DataProcessingException("Expected to insert at least one row, "
+                        + "but inserted 0 rows!");
+            }
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                book.setId(generatedKeys.getObject(1, Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't add new book: " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement statement = connection.prepareStatement(FIND_BY_ID)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(takeBook(resultSet));
```

Comment:
Instead of using both a constructor and setters, it's better to stick to one approach for consistency. Since the `takeBook` method uses setters, consider using a constructor with parameters in the `create` and `update` methods if a constructor is available, or use setters consistently across all methods.